src/test/java/org/jenkinsci/plugins/workflow/steps/EnvStepTest.java

```diff
@@ -83,6 +83,36 @@
         });
     }
 
+    @Test public void mapOverriding() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(
+                    "env.CUSTOM = 'initial'\n" +
+                        "env.FOOPATH = node {isUnix() ? '/opt/foos' : 'C:\\\\foos'}\n" +
```

Comment:
You need not copy all these exotic corner cases from overriding. Should suffice to have a simple script like

```groovy
withEnv([a: 1, b: 2]) {
  echo "a=$a b=$b"
}
```